From 34195b53c829a55584d61b1c8b8ca3eb620d95d7 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 26 Sep 2024 23:48:00 +0200
Subject: [PATCH] link error libraries

---
diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index f314d3a..71532f6 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -517,6 +517,7 @@ add_dependencies(scotch parser_yy_c parser_ll_c)
 if(NOT WIN32)
   target_link_libraries(scotch PUBLIC m)
 endif(NOT WIN32)
+target_link_libraries(scotch PUBLIC scotcherr)
 target_include_directories(scotch PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
@@ -772,6 +773,7 @@ if(BUILD_PTSCOTCH)
     SOVERSION ${SCOTCH_VERSION}.${SCOTCH_RELEASE}
     COMPILE_FLAGS -DSCOTCH_PTSCOTCH)
 
+  target_link_libraries(ptscotch PUBLIC ptscotcherr)
   target_link_libraries(ptscotch PUBLIC scotch MPI::MPI_C)
 
   target_include_directories(ptscotch PUBLIC
-- 
2.45.0

